azurerm/config.go

```diff
@@ -198,10 +198,16 @@ func withRequestLogging() autorest.SendDecorator {
 }
 
 func setUserAgent(client *autorest.Client) {
-	version := terraform.VersionString()
-	client.UserAgent = fmt.Sprintf("HashiCorp-Terraform-v%s", version)
+	tfVersion := fmt.Sprintf("HashiCorp-Terraform-v%s", terraform.VersionString())
 
-	// append the CloudShell version to the user agent
+	// if the user agent already has a value append the Terraform user agent string
+	if curUserAgent := client.UserAgent; curUserAgent != "" {
+		client.UserAgent = fmt.Sprintf("%s;%s", curUserAgent, tfVersion)
```

Comment:
Given this function is called in multiple places - have you tested this behaviour and checked the outputs?

Reply:
Yes I have, you can see the results [here](https://gist.github.com/jeffreyCline/ba277c2779cae20a72199f5cb6bf5f07). It works as expected without any duplicate strings added. However, I question why it gets called 164 times when provisioning a single object.

Reply:
👍 cool - AFAIK that's due to us setting the user agent for each API Client (such as a SQLDBClient/CosmosDBClient/AppServiceClient etc) that we instantiate when the Provider Plugin's initialised